tests/cases/conformance/types/rest/objectRest.ts

```diff
@@ -36,3 +36,5 @@ let computed = 'b';
 let computed2 = 'a';
 var { [computed]: stillNotGreat, [computed2]: soSo,  ...o } = o;
 ({ [computed]: stillNotGreat, [computed2]: soSo, ...o } = o);
+
+var noContextualType = ({ aNumber = 12, ...notEmptyObject }) => aNumber;
```

Comment:
can you add a test where notEmptyObject.somePropoerty is used, to make sure we get an error if the behavior changes, instead of just the type-baseline changes.